src/Serializer/SerializerContextBuilder.php

```diff
@@ -84,10 +84,6 @@ public function createFromRequest(Request $request, bool $normalization, ?array
             }
         }
 
-        if (null === $context['output'] && $this->getStateOptionsClass($operation)) {
-            $context['force_resource_class'] = $operation->getClass();
-        }
-
```

Comment:
@GromNaN this is probably fixing the issue you commented about the denormalizer. 

Could you review again?